teuthology/repo_utils.py

```diff
@@ -323,9 +335,9 @@ def fetch_repo(url, branch, bootstrap=None, lock=True):
             try:
                 while proceed():
                     try:
-                        enforce_repo_state(url, dest_path, branch)
+                        enforce_repo_state(url, dest_path, branch, commit)
                         if bootstrap:
-                            bootstrap(dest_path)
+                            bootstrap(dest_path, commit)
```

Comment:
The bootstrap has nothing to do with commit and only used to check it to proceed:

```python
if commit and os.path.exists(sentinel) or is_fresh(sentinel):
            log.info(
                "Skipping bootstrap as it was already done in the last %ss",
                FRESHNESS_INTERVAL,
            )
            return
```

We rather do this check outside of bootstrap method.